### PR TITLE
Remove self signer

### PR DESCRIPTION
**Pull request by obycode**
Note that for now, the blind-signer binary is completely untested and probably incomplete, but the main goal here is to remove the self-signer.

### CODE REVIEW

File: blind-signer/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    let reward_cycle = burnchain
+        .block_height_to_reward_cycle(tip.block_height)
+        .unwrap();
+    let prepare_phase_start = burnchain
+        .pox_constants
+        .prepare_phase_start(burnchain.first_block_height, reward_cycle);
+
+    if tip.block_height >= prepare_phase_start {
+        // If the key is already set, do nothing.
+        if is_key_set_for_cycle(reward_cycle + 1, conf.is_mainnet(), &conf.node.rpc_bind)
```
Comment (jferrant) [inline blind-signer/src/lib.rs:155]: Nit: bit easy to forget this +1 throughout. Would assign to a var called "next_reward_cycle" just so we don't accidentally remove one/forget this +1 where appropriate.

**Comment by jferrant** [comment]
I do wonder if this basically is just like running the current signer as a single signer especially since it still seems to use stackerdb/has the need to register a signer with the typical stack-stx function call unless I am missings omething?

**Comment by obycode** [comment]
Ok, I reverted the refactoring so now this PR just has the removal of the self-signer with the blind signing in the tests.

**Review by jferrant — DISMISSED** [review]
LGTM! :D

**Review by kantai — DISMISSED** [review]

**Review by jferrant — DISMISSED** [review]

**Review by zone117x — APPROVED** [review]

**Review by zone117x — APPROVED** [review]

**Review by jferrant — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Pull request merged**